conf.py

```diff
@@ -319,8 +319,12 @@ def xapian_code_example_command(ex):
     elif highlight_language == 'tcl':
         return "tclsh %s" % xapian_code_example_filename(ex)
     elif highlight_language == 'c++':
-        return "g++ `xapian-config --cxxflags` %s code/c++/support.cc -o %s `xapian-config --libs`\n./%s" \
-            % (xapian_code_example_filename(ex), ex, ex)
+        if option == 0:
+            return "g++ `xapian-config --cxxflags` %s code/c++/support.cc  -o %s `xapian-config --libs`\n./%s" \
+                % (xapian_code_example_filename(ex), ex, ex)
```

Comment:
You seem to have introduced an extra space before "-o" here (and then copied it below).